lib/Elastica/Search.php

```diff
@@ -475,7 +475,7 @@ public function count($query = '', $fullResult = false)
     {
         $this->setOptionsAndQuery(null, $query);
 
-        $query = $this->getQuery();
+        $query = clone $this->getQuery();
```

Comment:
Could you add a quick note (docs comment) on what it is cloned? Just to make sure nobody will remove it in the future ...